lib/scanner/csv-scanner/csv-scanner.c

```diff
@@ -396,15 +397,14 @@ csv_scanner_init(CSVScanner *scanner, CSVScannerOptions *options, const gchar *i
 {
   memset(scanner, 0, sizeof(*scanner));
   scanner->src = input;
-  scanner->current_value = g_string_sized_new(128);
+  scanner->current_value = scratch_buffers2_alloc();
   scanner->current_column = NULL;
```

Comment:
this is good, as csv_scanner_init()/deinit would be called in csv_parser_process() and the state is a local variable.